test/validation/api/pktio/pktio.c

```diff
@@ -2709,6 +2709,36 @@ static void pktio_test_chksum_in_ipv4(void)
 			  pktio_test_chksum_in_ipv4_test);
 }
 
+static void pktio_test_chksum_bad_in_ipv4_prep(odp_packet_t pkt)
+{
+	odp_u16sum_t chksum;
+	uint32_t offset;
+
+	odph_ipv4_csum_update(pkt);
+
+	/* Insert incorrect checksum */
+	offset = odp_packet_l3_offset(pkt);
+	odp_packet_copy_to_mem(pkt,
+					offset + ODPH_IPV4HDR_CSUM_OFFSET,
+					2, &chksum);
+	chksum ^= 1;
+	odp_packet_copy_from_mem(pkt,
+					offset + ODPH_IPV4HDR_CSUM_OFFSET,
+					2, &chksum);
```

Comment:
Indentation seems to be a little off here.

Reply:
Yes, indention should be aligned with parenthesis. Actually, this checksum modification could be done the same way as is done in UDP/STCP prep functions (without copies).

hdr = odp_packet_l3_ptr(pkt, NULL);
hdr->chksum ^= 1;

Copy is needed is packet is segmented in the middle of the field. I guess this test application makes sure, or can be sure that all headers fit into the first segment of the packet.

Reply:
Ack